frame/staking/src/tests.rs

```diff
@@ -440,13 +440,24 @@ fn no_candidate_emergency_condition() {
 			let res = Staking::chill(Origin::signed(10));
 			assert_ok!(res);
 
-			// trigger era
-			mock::start_active_era(1);
+			let current_era = CurrentEra::get();
 
-			// Previous ones are elected. chill is invalidates. TODO: #2494
+			// try trigger new era
+			mock::run_to_block(20);
+			assert_eq!(
+				*staking_events().last().unwrap(),
+				RawEvent::StakingElectionFailed,
+			);
+			// No new era is created
+			assert_eq!(current_era, CurrentEra::get());
+
+			// Go to far further session to see if validator have changed
+			mock::run_to_block(100);
+
+			// Previous ones are elected. chill is not effective in active era (as era hasn't changed)
 			assert_eq_uvec!(validator_controllers(), vec![10, 20, 30, 40]);
-			// Though the validator preferences has been removed.
-			assert!(Staking::validators(11) != prefs);
+			// The chill is still pending.
+			assert!(!<Staking as crate::Store>::Validators::contains_key(11));
```

Comment:
Are there any tests that document session + election_provider_multi_phase + stakig?

I would be interested in seeing a spec that shows a failed election, era extension, and then using `election_provider_multi_phase::ForceOrigin` to set a solution and succesfully start a new era. Could serve as a documented refference to show how to handle an election emergency.

Reply:
there isn't, I wanted to look at simnet to implement this kind of scenario, maybe some specific test could also be good.

Reply:
Would it be possible to make a test module with a mock runtime that includes those pallets and then just do a normal test that walks through it? If so I can make an issue just for that

Reply:
yes we can do it

Reply:
https://github.com/paritytech/substrate/issues/9057